core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go

```diff
@@ -217,21 +217,43 @@ func (r *EvmRegistry) allowedToUseMercury(opts *bind.CallOpts, upkeepId *big.Int
 		return encoding.NoPipelineError, encoding.UpkeepFailureReasonNone, false, allowed.(bool), nil
 	}
 
-	cfg, err := r.registry.GetUpkeepPrivilegeConfig(opts, upkeepId)
+	payload, err := r.packer.PackGetUpkeepPrivilegeConfig(upkeepId)
+	if err != nil {
+		// pack error, no retryable
+		r.lggr.Warnf("failed to pack getUpkeepPrivilegeConfig data for upkeepId %s: %s", upkeepId, err)
+
+		return encoding.RpcFlakyFailure, encoding.UpkeepFailureReasonNone, true, false, fmt.Errorf("failed to pack upkeepId: %w", err)
```

Comment:
will need to fix this error instead of RpcFlakyFailure

Reply:
I can create a new error type that makes more sense in this case.

Reply:
This might fit in here but best to confirm with felix https://github.com/smartcontractkit/chainlink/blob/develop/core/services/ocr2/plugins/ocr2keeper/evm21/encoding/interface.go#L39

Reply:
agree that this should be `PackUnpackDecodeFailed`

Reply:
and this should be not retryable

Reply:
Ok. How about `PackUnpackDecodeFailed`? And not retryable like Felix said below.